pkg/operator/staticpod/startupmonitor/startup_monitor.go

```diff
@@ -0,0 +1,370 @@
+package startupmonitor
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+)
+
+// HealthChecker a contract between the startup monitor and operators
+// allows for initialization and an operand's health condition assessment
+type HealthChecker interface {
+	// Start is used to initialize and start the underlying health monitor
+	Start(context.Context) error
+
+	// IsTargetHealthy defines a function that abstracts away assessing operand's health condition.
+	// the provided functions should be async and cheap in a sense that it shouldn't assess the target
+	// only read the current state.
+	// mainly because we acquire a lock on each sync.
+	IsTargetHealthy() (healthy bool, reason string, err error)
```

Comment:
What is err? Is it nil when unhealthy? Where do we get a human readible message from?

Reply:
I was thinking that the reason would be `fallback-reason` and the `err` could be `fallback-message`

```
   - `startup-monitor.static-pods.openshift.io/fallback-reason: MachineReadableReason`
   - `startup-monitor.static-pods.openshift.io/fallback-message: "Human readable, possibly multiline message"`
```

Reply:
err is normally a real error. Stroustrup said somewhere: don't use error flow for non-error cases.

Reply:
do you think we need `IsTargetHealthy() (healthy bool, reason string, msg string, err error)` ?

Reply:
or just `IsTargetHealthy() (reason string, msg string, err error)` ?

Reply:
would keep the bool, but that's just taste.